httpsender/Capture and Replace Anti CSRF Token.js

```diff
@@ -0,0 +1,101 @@
+    var numberOfParameterTypes = parameterTypesList.length;
+    for (var index=0; index < numberOfParameterTypes; index++) {
+        if (parameterTypesList[index] != null && parameterTypesList[index] === formParamType) {
+            var formParams = msg.getFormParams();
+            // println ("Form Params before update: " + formParams);
+            var updatedFormParams = modifyParams(formParamType, formParams);
+            // println ("Form Params after update: " + updatedFormParams);
+            msg.setFormParams(updatedFormParams);
+        } else if (parameterTypesList[index] != null && parameterTypesList[index] === urlParamType) {
+            var urlParams = msg.getUrlParams();
+            // println ("Url Params before update: " + urlParams);
+            var updatedUrlParams = modifyParams(urlParamType, urlParams);
+            // println ("Url Params after update: " + updatedUrlParams);
+            msg.setGetParams(updatedUrlParams);
+        } else if (parameterTypesList[index] != null && parameterTypesList[index] === cookieParamType) {
+            var cookieParams = msg.getCookieParams();
+            // println ("Cookie Params before update: " + cookieParams);
+            var updatedCookieParams = modifyParams(cookieParamType, cookieParams);
+            // println ("Cookie Params after update: " + updatedCookieParams);
+            msg.setCookieParams(updatedCookieParams);
+        }
+    }
+}
+
+function responseReceived(msg, initiator, helper) {
+    // println('responseReceived called for url=' + msg.getRequestHeader().getURI().toString())
+    // println ('Initiator: ' + initiator)
+    if (msg.getRequestHeader().getURI().toString().match(urlRegEx) != null) {
+        var csrfTokenValue = msg.getResponseBody().toString().match(csrfTokenValueRegEx);
+        if (csrfTokenValue != null && csrfTokenValue.length > matcherGroupNumber) {
+            println('Latest CSRF Token value: ' + csrfTokenValue[matcherGroupNumber]);
+            org.zaproxy.zap.extension.script.ScriptVars.setGlobalVar("anti.csrf.token.value", csrfTokenValue[matcherGroupNumber]);
+        }
+    }
+}
+
+function modifyParams(parameterType, params) {
+    var iterator = params.iterator();
+    while(iterator.hasNext()) {
+        var param = iterator.next();
+        // Check if the url parameters has the antiCsrfTokenName in it.
+        if (param.getName().equals(antiCsrfTokenName)) {
+            var secureTokenValue = param.getValue();
+            // Check for the value of AntiCsrfTokenName in the existing request with the latest value captured from previous requests.
+            if (!secureTokenValue.equals(org.zaproxy.zap.extension.script.ScriptVars.getGlobalVar("anti.csrf.token.value"))) {
+                // Remove the antiCsrfToken from the form parameters.
+                var oldCsrfToken = new org.parosproxy.paros.network.HtmlParameter(parameterType, antiCsrfTokenName, secureTokenValue);
+                params.remove(oldCsrfToken);
```

Comment:
This and the above statement could be replaced with ```iterator.remove();```.
Or, since it's just being changed the value of the parameter the above and following two statements could be replaced with:
```JavaScript
param.setValue(org.zaproxy.zap.extension.script.ScriptVars.getGlobalVar("anti.csrf.token.value"))
```
It also makes the parameter ```parameterType``` obsolete.